src/DockWidgetSideTab.h

```diff
@@ -0,0 +1,168 @@
+#ifndef DockWidgetSideTabH
+#define DockWidgetSideTabH
+/*******************************************************************************
+** Qt Advanced Docking System
+** Copyright (C) 2017 Uwe Kindler
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License as published by the Free Software Foundation; either
+** version 2.1 of the License, or (at your option) any later version.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+** Lesser General Public License for more details.
+**
+** You should have received a copy of the GNU Lesser General Public
+** License along with this library; If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+
+//============================================================================
+/// \file   DockWidgetTab.h
+/// \author Syarif Fakhri
+/// \date   05.09.2022
+/// \brief  Declaration of CDockWidgetSideTab class
+//============================================================================
+
+//============================================================================
+//                                   INCLUDES
+//============================================================================
+#include <QFrame>
+
+#include "ads_globals.h"
+
+namespace ads
+{
+struct DockWidgetSideTabPrivate;
+class CDockWidget;
+class CSideTabBar;
+class CDockWidgetTab;
+struct SideTabIconLabelPrivate;
+
+/**
+ * A dock widget Side tab that shows a title or an icon.
+ * The dock widget tab is shown in the side tab bar to switch between
+ * pinned dock widgets
+ */
+class ADS_EXPORT CDockWidgetSideTab : public QFrame
+{
+    Q_OBJECT
+
+    Q_PROPERTY(SideTabBarArea sideTabBarArea READ sideTabBarArea)
+	Q_PROPERTY(QSize iconSize READ iconSize WRITE setIconSize)
+
+private:    
+	DockWidgetSideTabPrivate* d; ///< private data (pimpl)
+    friend struct DockWidgetSideTabPrivate;
+	friend class CDockWidget;
+	friend class COverlayDockContainer;
+
+protected:
+	friend class CSideTabBar;
+	friend class CDockAreaWidget;
+	friend class CDockContainerWidget;
+
+	void mousePressEvent(QMouseEvent* event) override;
+
+	void setSideTabBar(CSideTabBar *SideTabBar);
+	void removeFromSideTabBar();
+
+public:
+    using Super = QFrame;
+
+    /**
+     * Dock widget side tab bar locations
+     */
+    enum SideTabBarArea
+    {
+        LeftTop,
+        LeftBottom,
+        RightTop,
+        RightBottom,
+		Bottom
```

Comment:
Would be great, if later a Top area could be added

Reply:
![image](https://user-images.githubusercontent.com/4475295/195859170-a9e9d5ca-def5-47f4-83c8-650765ad6848.png)

Added a top bar area!

Reply:
FYI: The calculation of which tab bar area it should go to is in: `CDockContainerWidget::getDockAreaPosition`, but the general idea is, if width > height, it'll go to the top or bottom based on distance. Otherwise, left or right based on distance.